### ISSUE TITLE
Enhancement: kpm should check the kcl compiler version used and update it into 'kcl.mod'

### ISSUE THREAD

**Issue by zong-zhe** [kcl-lang/kpm#186]
## Enhancement
At present, the version of kcl compiler used by kpm is hardcoded, and the compiler version needs to be specified through the field `edition` in section `[package]` kcl.mod.

**Comment by AkashKumar7902** [comment]
Hey @zong-zhe @Peefy,
can I work on this issue ? I really want to start contributing so that i can understand the codebase better.

**Comment by Peefy** [comment]
> Hey @zong-zhe @Peefy, can I work on this issue ? I really want to start contributing so that i can understand the codebase better.

OK, I've assigned this issue to you! If you have any questions, please feel free to communicate with me at any time.

**Comment by Peefy** [comment]
You're right, I'm surprised you found the accurate version dependency so quickly.

**Comment by AkashKumar7902** [comment]
@Peefy does kcl run in vm (guessing by the filename) ? also bit confused about the fact that version in that file spells v0.7.0 but my compiler is 0.7.4.

**Comment by zong-zhe** [comment]
Hi ! @AkashKumar7902 😄

Thanks for your contributions to KCL.

I think there maybe some misunderstanding for https://github.com/kcl-lang/kpm/issues/186. This issue is just a record of a feature, without further design. Let me add more details. 

This feature refers to rust's edition fields:

More details about edition in cargo.toml - https://doc.rust-lang.org/cargo/reference/manifest.html#the-edition-field

In the ideal state, the work of this issue should be divided into at least two parts.

1. Creating a package: When initializing a KCL package, the corresponding version number should be added to the edition field in `kcl.mod` based on the version of the KCL compiler currently called by `kpm`.

1. During the compilation process, check the edition of the current KCL package and the edition field of the third-party dependencies that the package depends on, and choose the appropriate KCL compiler for compilation based on the edition field.

However, given the current situation of kpm and KCL projects, the workload of implementing the second point in the above process is too large. For the domain-specific language KCL, the actual benefits of this the second point are too small, and this feature does not need to align with the Rust. Therefore, we can temporarily implement the first part of the content for now.

1. When using the init command to initialize a new KCL package, set the corresponding KCL compiler version according to the KCLVM version used by the current kpm package manager.
